pkg/internal/kube/store.go

```diff
@@ -289,17 +306,36 @@ func (s *Store) serviceNameNamespaceForMetadata(om *informer.ObjectMeta) (string
 	} else {
 		name, namespace = s.serviceNameNamespaceForOwner(om)
 	}
-	if s.sourceLabels.ServiceName != "" {
-		if on, ok := om.Labels[s.sourceLabels.ServiceName]; ok {
-			name = on
+	if nameFromMeta := s.valueFromMetadata(om,
+		s.metadataSources.ServiceNameAnnotations,
+		s.metadataSources.ServiceNameLabels,
+	); nameFromMeta != "" {
+		name = nameFromMeta
+	}
+	if nsFromMeta := s.valueFromMetadata(om,
+		s.metadataSources.ServiceNamespaceAnnotations,
+		s.metadataSources.ServiceNamespaceLabels,
+	); nsFromMeta != "" {
+		namespace = nsFromMeta
+	}
+	return name, namespace
```

Comment:
I think the fallback values are missing (at least not in this PR) - which are maybe the most important part.

In the operator, those are:
- name: https://github.com/open-telemetry/opentelemetry-operator/blob/84200196ff4e0fc9d8999b31891263f093fe7052/pkg/instrumentation/sdk.go#L408-L434
- version: https://github.com/open-telemetry/opentelemetry-operator/blob/84200196ff4e0fc9d8999b31891263f093fe7052/pkg/instrumentation/sdk.go#L456-L467
- instance.id: https://github.com/open-telemetry/opentelemetry-operator/blob/84200196ff4e0fc9d8999b31891263f093fe7052/pkg/instrumentation/sdk.go#L486-L490

Reply:
In the Beyla, the fallback values are previously set into the `name` and `namespace` variables, then overridden here if `valueFromMetadata` returns a non-empty value.

That's the reason I was mentioning that moving this code into a shared library could be complex due to the way multiple components could populate/store the same information.

Reply:
I understand - the important part is to have the same logic

- can you point me to the logic?
- can you also implement the logic for version?

Reply:
On one side, Beyla does not store the metadata from the Pod owners (DaemonSet, Deployment, Job, etc...) so we cannot query the fallbacks one by one.

Instead, Beyla infers the pod owner name from the actual pod name (assuming the kubernetes pod naming conventions). So at the point we execute the above, service.name already contains the value of of one of the fallbacks (which at the end are nothing that the Pod owner name).

Is then that, if the service name/namespace can't be inferred from env/annotations/labels (in that priority), the original value is kept (the fallback).

Reply:
so it should result in the same outcome - correct?

Reply:
Yes. My initial concern was not about the outcome but the code reutilization.